src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -136,7 +136,7 @@ walletApplication
     -> m Application
 walletApplication serv = do
     wsConn <- getWalletWebSockets
-    serv >>= return . upgradeApplicationWS wsConn . serve walletApi
+    upgradeApplicationWS wsConn . serve walletApi <$> serv
```

Comment:
Or, sure, you're right. Going from left to right and then from right to left is horrible. Didn't notice that `(.)` somehow. In that case I agree with you here.